File: SiteVisitsNetCore2020Web/ClientApp/src/app/models/VisitFlat.ts
export class VisitFlat {
  id: string;
  visitDatetime: Date;
  numberOfTimes: number;
  city: string;
  region: string;
  country: string;
  postalCode: string;
  location: string;
  logFileName: string;
  visitType: string;
  ipAddress: string;
  isp: string;
  //extractedTerms: Array<string>;
  pageUrl: string;
  //pageUrlVariation: string;
  pageTitle: string;
  //pageTitleVariation: PageTitleVariation;
  cameFrom: string;
  combinedTerms: string;
  /****
      public virtual Browser Browser { get; set; }
      public virtual Device Device { get; set; }
      public virtual SearchEngine SearchEngine { get; set; }
      ***/

  constructor(options: {
    id: string, visitDatetime: string,
    numberOfTimes: number,
    seTerm: string,
    logFileName: string,
    visitType: string,
    ipAddress: string,
    isp: string,
    city: string,
    region: string,
    country: string,
    postalCode: string,
    location: string,
    pageUrl: string,
    pageTitle: string,
    cameFrom: string,
    //extractedTerms: Array<string>
    combinedTerms: string
  }) {
    this.id = options.id;
    this.visitDatetime = new Date(options.visitDatetime);
    this.numberOfTimes = options.numberOfTimes;
    this.logFileName = options.logFileName;
    this.visitType = options.visitType;
    this.ipAddress = options.ipAddress;
    this.isp = options.isp;
    this.city = options.city;
    this.region = options.region;
    this.country = options.country;
    this.postalCode = options.postalCode;
    this.location = options.location;
    //this.extractedTerms = options.extractedTerms;
    this.pageUrl = options.pageUrl;
    this.pageTitle = options.pageTitle;
    this.cameFrom = options.cameFrom;
    this.combinedTerms = options.combinedTerms;
  }
}

File: SiteVisitsNetCore2020Web/Models/VisitFlat.cs
using System;

namespace SiteVisitsNetCore2020Web.Models
{
    public class VisitFlat
    {
        public Guid Id { get; set; }
        public DateTime VisitDatetime { get; set; }

        public string IpAddress { get; set; }
        public int NumberOfTimes { get; set; }
        public string Organization { get; set; }
        public string City { get; set; }
        public string Region { get; set; }
        public string Country { get; set; }
        public string PostalCode { get; set; }
        public string Location { get; set; }
        public string PageUrl { get; set; }
        public string PageTitle { get; set; }
        public string CameFrom { get; set; }
        public string CombinedTerms { get; set; }


    }
}
